### ISSUE TITLE
[Bug Report] Memory leak on route change with a visible tooltip

### ISSUE THREAD

**Issue by syn-zeta** [ElemeFE/element#7715]
### Element UI version
1.4.8

### OS/Browsers version
Mac OS X 10.11.6 / Chrome 61.0.3163.100 (Official Build) (64-bit)

### Vue version
2.5.2

### Reproduction Link
[https://jsfiddle.net/kydfpodc/1/show/](https://jsfiddle.net/kydfpodc/1/show/)

### Steps to reproduce
- Click on "go to Foo" to load the table initially
- Decrease viewport width until the address column in the table has ellipsis, and shows tooltips on hover
- Click on one of the rows in the address column while the tooltip is visible to be redirected to "Bar"
- Click on "go to Foo" to view the table again

### What is Expected?
Node count after step 4 to be the same as the node count after step 1.

### What is actually happening?
Steadily increasing node count as the detached tables remain in memory due to references held in tooltips.

![image](https://user-images.githubusercontent.com/26761352/31988505-f94dd95a-b98c-11e7-8402-2f786bc5acb8.png)

![image](https://user-images.githubusercontent.com/26761352/31988489-eb504b12-b98c-11e7-9fff-8bf13ac27b4a.png)

**Comment by syn-zeta** [comment]
There also seems to be an issue with memory allocations as there are frequent GC visible in the HEAP section in the first screenshot.

**Comment by syn-zeta** [comment]
@QingWei-Li @Leopoldthecoder I did some investigation and found that the reference causing the leak is the `this` binding of `update` method in `popper.js`. Basically, the bottom line in the second screenshot above:

![image](https://user-images.githubusercontent.com/26761352/35965835-f02d786c-0c70-11e8-8213-674e89380501.png)

If we comment out line 456, and update 457, 465 to `root.addEventListener('<event>', this.update);` (and adjust `_removeEventListeners` similarly as well), the leak will be fixed, although the update method will error out on scroll/resize.

I've tried a few things to avoid the `this` binding, but haven't figured out a proper solution yet. Just thought I'll share my findings here in case you guys have any ideas.
